v2/yarpcthrift/thriftrw-plugin-yarpcv2/roundtrip_test.go

```diff
@@ -299,27 +274,13 @@ func testRoundTrip(t *testing.T, enveloped, multiplexed bool) {
 			require.Equal(t, reflect.Interface, clientType.Kind(),
 				"invalid test: newClientFunc must return an Interface")
 
-			// The following blob is equivalent to,
-			//
-			// 	var clientHolder struct {
-			// 		Client ${service}client.Interface `service:"roundtrip-server"`
-			// 	}
-			// 	yarpc.InjectClients(dispatcher, &clientHolder)
-			// 	client := clientHolder.Client
-			//
-			// Optionally with the `thrift:"..."` tag if thriftTag was
-			// non-empty.
-			structType := reflect.StructOf([]reflect.StructField{
-				{
-					Name: "Client",
-					Type: clientType,
-					Tag:  reflect.StructTag(`service:"roundtrip-server"` + thriftTag),
-				},
-			})
-			clientHolder := reflect.New(structType).Elem()
-			yarpc.InjectClients(dispatcher, clientHolder.Addr().Interface())
-			client := clientHolder.Field(0)
-			assert.NotNil(t, client.Interface(), "InjectClients did not provide a client")
+			clientArgs := []reflect.Value{reflect.ValueOf(clientD)}
+			for _, opt := range clientOpts {
+				clientArgs = append(clientArgs, reflect.ValueOf(opt))
+			}
+
+			client := reflect.ValueOf(tt.newClientFunc).
```

Comment:
so much reflection :O

Should we check the last `arg` for what I think should have an `error` and make sure it's nil?